### PR TITLE
Store access token missing

### PR DESCRIPTION
**Pull request by ph**
This PR add a few things

    - It takes the keystore encryption implementation and create an
      io.Reader and io.Writer for it, we currently work with variable lenght
      blocks and we are not compatible for now with the one in libbeat.

    - It implements a few storage mechanism and wrapper:
      - DiskStore: Save a io.reader directly to disk, the content is saved
        in a temporary file and the target is replace in an atomic
        operation.
      - EncryptedDiskStore: Same as Keystore but use the encrypted io.Reader
        and io.Writer.

    - A fleet configuration is created in _meta/agent.fleet.yml, the content
      of this file is packed in the binary. When a user enroll the agent,
      we do a backup of the current agent.yml and the content is replaced
      with the agent.fleet.yml

    - The enrollment information is saved into a "fleet.yml"

    - The managed mode now read the content of the fleet.yml and creates a
      Kibana API client.

    - A Separates Fleet Config struct is created.

    - Enroll will now ask for confirmation before replacing the user
      configuration.

ref: #14951

### CODE REVIEW

**Comment by ph** [comment]
@michalpristas 

Have you seen this error before CI?
```
-- FAIL: TestShortRun (0.00s)

    operator_test.go:36: operation-start: chown /home/travis/gopath/src/github.com/elastic/beats/x-pack/agent/pkg/agent/operation/tests/scripts/short--1.0.yml: operation not permitted

=== RUN   TestShortRunInvalid

2019-12-12T20:55:33Z INFO	operation_fetch.go:80	operation 'operation-fetch' downloaded bumblebee. into 

2019-12-12T20:55:33Z INFO	operator.go:199	operation 'operation-verify' skipped for bumblebee.

2019-12-12T20:55:33Z INFO	process.go:72	address assigned to the process '/bin/bumblebee/bumblebee': '127.0.0.1:10000'

--- PASS: TestShortRunInvalid (1.00s)

=== RUN   TestLongRunWithStop

2019-12-12T20:55:34Z INFO	operation_fetch.go:80	operation 'operation-fetch' downloaded long.1.0 into 

2019-12-12T20:55:34Z INFO	operator.go:199	operation 'operation-verify' skipped for long.1.0

2019-12-12T20:55:34Z INFO	operator.go:199	operation 'operation-install' skipped for long.1.0

2019-12-12T20:55:34Z INFO	process.go:72	address assigned to the process '/bin/sh': '127.0.0.1:10000'

--- FAIL: TestLongRunWithStop (0.00s)

    operator_test.go:71: operation-start: failed to start '/bin/sh': fork/exec /bin/sh: operation not permitted

=== RUN   TestLongRunWithCrash

2019-12-12T20:55:34Z INFO	operation_fetch.go:80	operation 'operation-fetch' downloaded long.1.0 into 

2019-12-12T20:55:34Z INFO	operator.go:199	operation 'operation-verify' skipped for long.1.0

2019-12-12T20:55:34Z INFO	operator.go:199	operation 'operation-install' skipped for long.1.0

2019-12-12T20:55:34Z INFO	process.go:72	address assigned to the process '/bin/sh': '127.0.0.1:10000'

--- FAIL: TestLongRunWithCrash (0.01s)

    operator_test.go:112: operation-start: failed to start '/bin/sh': fork/exec /bin/sh: operation not permitted

=== RUN   TestTwoProcesses

2019-12-12T20:55:34Z INFO	operation_fetch.go:80	operation 'operation-fetch' downloaded long.1.0 into 

2019-12-12T20:55:34Z INFO	operator.go:199	operation 'operation-verify' skipped for long.1.0

2019-12-12T20:55:34Z INFO	operator.go:199	operation 'operation-install' skipped for long.1.0

2019-12-12T20:55:34Z INFO	process.go:72	address assigned to the process '/bin/sh': '127.0.0.1:10000'

--- FAIL: TestTwoProcesses (0.00s)

    operator_test.go:160: operation-start: failed to start '/bin/sh': fork/exec /bin/sh: operation not permitted

=== RUN   TestConfigurableRun

2019-12-12T20:55:34Z INFO	operation_fetch.go:80	operation 'operation-fetch' downloaded configurable.1.0 into 

2019-12-12T20:55:34Z INFO	operator.go:199	operation 'operation-verify' skipped for configurable.1.0

2019-12-12T20:55:34Z INFO	operator.go:199	operation 'operation-install' skipped for configurable.1.0

2019-12-12T20:55:36Z INFO	process.go:72	address assigned to the process '/home/travis/gopath/src/github.com/elastic/beats/x-pack/agent/pkg/agent/operation/tests/scripts/configurable-1.0-darwin-x86/configurable': '127.0.0.1:10000'

--- FAIL: TestConfigurableRun (2.51s)

    operator_test.go:209: found file /home/travis/gopath/src/github.com/elastic/beats/x-pack/agent/pkg/agent/operation/tests/scripts/configurable-1.0-darwin-x86/configurable

    operator_test.go:214: operation-start: failed to start '/home/travis/gopath/src/github.com/elastic/beats/x-pack/agent/pkg/agent/operation/tests/scripts/configurable-1.0-darwin-x86/configurable': fork/exec /home/travis/gopath/src/github.com/elastic/beats/x-pack/agent/pkg/agent/operation/tests/scripts/configurable-1.0-darwin-x86/configurable: operation not permitted

=== RUN   TestConfigurableByFileRun

2019-12-12T20:55:36Z INFO	operation_fetch.go:80	operation 'operation-fetch' downloaded configurablebyfile.1.0 into 

2019-12-12T20:55:36Z INFO	operator.go:199	operation 'operation-verify' skipped for configurablebyfile.1.0

2019-12-12T20:55:36Z INFO	operator.go:199	operation 'operation-install' skipped for configurablebyfile.1.0

--- FAIL: TestConfigurableByFileRun (0.00s)

    operator_test.go:294: found file /home/travis/gopath/src/github.com/elastic/beats/x-pack/agent/pkg/agent/operation/tests/scripts/configurablebyfile-1.0-darwin-x86/configurablebyfile

    operator_test.go:299: operation-start: chown /home/travis/gopath/src/github.com/elastic/beats/x-pack/agent/pkg/agent/operation/tests/scripts/configurablebyfile--1.0.yml: operation not permitted
```

**Comment by michalpristas** [comment]
seems like a permission problem. i've seen this running in a incorrectly setup VM, root priviledges are missing so you can `chown` or spawn a new process `bin/sh`

File: x-pack/agent/dev-tools/cmd/buildfleetcfg/buildfleetcfg.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```
Comment (michalpristas) [inline x-pack/agent/dev-tools/cmd/buildfleetcfg/buildfleetcfg.go]: 👍

File: x-pack/agent/dev-tools/cmd/buildfleetcfg/buildfleetcfg.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	if len(input) == 0 {
+		fmt.Fprintln(os.Stderr, "Invalid input source")
+		os.Exit(1)
+	}
+
+	l, err := licenses.Find(license)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "problem to retrieve the license, error: %+v", err)
+		os.Exit(1)
+		return
```
Comment (michalpristas) [inline x-pack/agent/dev-tools/cmd/buildfleetcfg/buildfleetcfg.go]: `return` not needed

File: x-pack/agent/dev-tools/cmd/buildfleetcfg/buildfleetcfg.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	data, err := gen(input, l)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error while generating the file, err: %+v\n", err)
+		os.Exit(1)
+	}
+
+	if output == "-" {
+		os.Stdout.Write(data)
+		return
+	} else {
```
Comment (michalpristas) [inline x-pack/agent/dev-tools/cmd/buildfleetcfg/buildfleetcfg.go]: style: no need for else, if condition is true there is a return inside

File: x-pack/agent/dev-tools/cmd/buildfleetcfg/buildfleetcfg.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+		Pack    string
+		Files   []string
+		License string
+	}{
+		Pack:    pack,
+		Files:   files,
+		License: l,
+	})
+
+	formatted, err := format.Source(buf.Bytes())
```
Comment (michalpristas) [inline x-pack/agent/dev-tools/cmd/buildfleetcfg/buildfleetcfg.go]: we dont wrap or enhance error so we can return here without additional if statement

File: x-pack/agent/pkg/core/plugin/app/process_cred.go [hunk 5]
```diff
@@ -2,7 +2,7 @@
 // or more contributor license agreements. Licensed under the Elastic License;
 // you may not use this file except in compliance with the Elastic License.
 
-// +build linux darwin
+// +build (linux OR darwin)
```
Comment (michalpristas) [inline x-pack/agent/pkg/core/plugin/app/process_cred.go]: i think previous version is correct

File: x-pack/agent/pkg/core/plugin/app/process_cred_other.go [hunk 6]
```diff
@@ -2,8 +2,7 @@
 // or more contributor license agreements. Licensed under the Elastic License;
 // you may not use this file except in compliance with the Elastic License.
 
-// +build !linux
-// +build !darwin
+// +build !linux AND !darwin
```
Comment (michalpristas) [inline x-pack/agent/pkg/core/plugin/app/process_cred_other.go]: same

**Comment by michalpristas** [comment]
check if you are the owner of a directory

**Comment by michalpristas** [comment]
@ph if you fix // +build directives as suggested build will pass

**Comment by ph** [comment]
@michalpristas Just pushed the commits for the previous comments, let's wait for the CI. :crossed_fingers:

**Pull request merged**